modules/sharedaddy/recaptcha.php

```diff
@@ -0,0 +1,168 @@
+		$this->secret_key = $secret_key;
+		$this->config     = wp_parse_args( $config, $this->get_default_config() );
+	}
+
+	/**
+	 * Get default config for this reCAPTCHA instance.
+	 *
+	 * @return array Default config
+	 */
+	public function get_default_config() {
+		return array(
+			'language'       => get_locale(),
+			'tag_class'      => 'g-recaptcha',
+			'tag_attributes' => array(
+				'theme'    => 'light',
+				'type'     => 'image',
+				'tabindex' => 0,
+			)
+		);
+	}
+
+	/**
+	 * Calls the reCAPTCHA siteverify API to verify whether the user passes
+	 * CAPTCHA test.
+	 *
+	 * @param string $response  The value of 'g-recaptcha-respones' in the submitted
+	 *                          form.
+	 * @param string $remote_ip The end user's IP address.
+	 *
+	 * @return bool|WP_Error Returns true if verified. Otherwise WP_Error is returned.
+	 */
+	public function verify( $response, $remote_ip ) {
+		// No need make a request if response is empty.
+		if ( empty( $response ) ) {
+			return new WP_Error( 'missing-input-response', $this->error_codes['missing-input-response'], 400 );
+		}
+
+		$resp = wp_remote_post( self::VERIFY_URL, $this->get_verify_request_params( $response, $remote_ip ) );
+		if ( is_wp_error( $resp ) ) {
+			return $resp;
+		}
+
+		$resp_decoded = json_decode( wp_remote_retrieve_body( $resp ), true );
+		if ( ! $resp_decoded ) {
+			return new WP_Error( 'invalid-json', 'Invalid JSON', 400 );
+		}
+
+		if ( isset( $resp_decoded['success'] ) && true === $resp_decoded['success'] ) {
+			return true;
+		}
+
+		$error_message = 'Unexpected response';
+		$error_code    = 'unexpected-response';
+
+		// Use the first error code if exists.
+		if ( isset( $resp_decoded['error-codes'] ) && is_array( $resp_decoded['error-codes'] ) ) {
+			if ( isset( $resp_decoded['error-codes'][0] ) && isset( $this->error_codes[ $resp_decoded['error-codes'][0] ] ) ) {
+				$error_message = $this->error_codes[ $resp_decoded['error-codes'][0] ];
+				$error_code    = $resp_decoded['error-codes'][0];
+			}
+		}
+
+		return new WP_Error( $error_code, $error_message, 400 );
+	}
+
+	/**
+	 * Get siteverify request parameters.
+	 *
+	 * @param string $response  The value of 'g-recaptcha-respones' in the submitted
+	 *                          form.
+	 * @param string $remote_ip The end user's IP address.
+	 *
+	 * @return array
+	 */
+	public function get_verify_request_params( $response, $remote_ip ) {
+		return array(
+			'body' => array(
+				'secret'   => $this->secret_key,
+				'response' => $response,
+				'remoteip' => $remote_ip,
+			),
+			'sslverify' => true,
+		);
+	}
+
+	/**
+	 * Get reCAPTCHA HTML to render.
+	 *
+	 * @return string
+	 */
+	public function get_recaptcha_html() {
+		return sprintf(
+			'
+			<div
+				class="%s"
+				data-sitekey="%s"
+				data-theme="%s"
+				data-type="%s"
+				data-tabindex="%s"></div>
+			<script type="text/javascript" src="https://www.google.com/recaptcha/api.js?hl=%s"></script>
```

Comment:
I'm not sure. However, I modified that to be opt-in in config (via constructor). The default is `async`. See https://github.com/gedex/jetpack/commit/a2f2ec729c5c8edf070dbb5a4fe6baf2f0f5d6e8.